Add BF16 type to _autocast_to_full_precision

yusuo opened the pull request:
Summary:
https://github.com/pytorch/pytorch/pull/63939/files has added FP16 support to torchscript.

This is to add BF16 device type when doing full conversion.

Test Plan: Unit test. Also tested BF16 locally on A100 using MLP model.

Differential Revision: D32027152

eellison commented:
Thanks! Maybe we can parameterize one of the tests in https://github.com/pytorch/pytorch/blob/master/test/test_jit_autocast.py to also loop over bfloat16. cc @jjsjann123

yusuo commented:
> Thanks! Maybe we can parameterize one of the tests in https://github.com/pytorch/pytorch/blob/master/test/test_jit_autocast.py to also loop over bfloat16. cc @jjsjann123

BF16 needs A100 hardware support. We probably cannot run those tests on non-A100 GPU.

jjsjann123 reviewed (approved):
Thanks a lot for patching this!

yusuo commented:
Added a unit test for BF16. Also modified the FP16 test to cover both side conversions.

yusuo committed:
Add BF16 type to _autocast_to_full_precision (#67707)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/67707

https://github.com/pytorch/pytorch/pull/63939/files has added FP16 support to torchscript.

This is to add BF16 device type when doing full conversion.

Test Plan: Unit test. Also tested BF16 locally on A100 using MLP model.

Reviewed By: idning

Differential Revision: D32027152

fbshipit-source-id: e8966dce7e892b5ec8ba5ef0e05fbd324c00f019
